http/client.go

```diff
@@ -427,6 +434,10 @@ func (c *Client) Peerstore() datastore.DSBatching {
 	panic("client side database")
 }
 
+func (c *Client) ACPModule(ctx context.Context) (immutable.Option[acp.ACPModule], error) {
+	panic("client side database")
```

Comment:
todo: This feels very wrong, and breaks the `client` interface.  It also means that users can only ever upload a policy and never do anything with it via http/cli.   This is a hard blocker for me, so do open a discord discussion if you disagree.

Reply:
> todo: This feels very wrong, and breaks the `client` interface.

What do you mean by that? which interface?


> It also means that users can only ever upload a policy and never do anything with it via http/cli.

That would be correct for what is introduced in this PR. The user can **only** add the policy to acp, the rest (registering, and checking) is managed by the application (DefraDB).

I see the user directly interacting with ACP Module as somewhat problematic (with the exception of bare minimum stuff that needs to exist to mimic sourcehub, i.e. the adding of policy for now).


> This is a hard blocker for me, so do open a discord discussion if you disagree.

I don't fully understand yet why not disclosing the underlying acp module to the user is a blocker? Can start the discussion once I understand the respective stance.


> It also prevents the proper testing of the public functions on `acp.ACPModule`.

Every functionality within `acp.ACPModule` is being tested through the integration tests. They are also possible to be testet through unit tests more directly (I am not a huge fan of unit tests but do plan on adding some unit tests).

Reply:
> which interface?

The interfaces in the client package.  By panicing instead of implementing it you are breaking the interface.

> I see the user directly interacting with ACP Module as somewhat problematic

Then please fix it :)

> Every functionality within acp.ACPModule is being tested through the integration tests.

Indirectly, via internal calls.  They are public though, and need to be tested directly.

Reply:
> The interfaces in the client package. By panicing instead of implementing it you are breaking the interface.

In that case the client is already __broken__ as `Rootstore()` `Datastore()`, `Headstore()` do the same.

I thought this was how the client limited certain functionality as there are so many other methods that are also panicking similarly that I would classify under the same use case as `ACPModule()`, which is essentially the acp store, like the `Rootstore()`, `Datastore()`, `Headstore()`.

Nonetheless, I am looking to see if I can omit `ACPModule()` completely from there but instead pass it around where it is needed through function arguments.


> > I see the user directly interacting with ACP Module as somewhat problematic
> 
> Then please fix it :)

I will try to remove it completely from there, but we should in that case also fix the other methods on the client that make the client broken, as someone else tomorrow might assume this is "ok" behavior.


> > Every functionality within acp.ACPModule is being tested through the integration tests.
> 
> Indirectly, via internal calls. They are public though, and need to be tested directly.

Are you saying this regarding `client.Collection` functions or `acp.ACPModule` functions? The latter as mentioned I will add unit tests for soon.

- [ ] Add unit tests for `acp.ACPModule` funcs

Reply:
> In that case the client is already broken as Rootstore() Datastore(), Headstore() do the same.

Yes.  That does not mean we should make it more broken though, especially for stuff that has very little reason to be broken (of course exposing an IPFS store via the CLI is a lot of low-priority hassle, and those funcs also existed before the refactor that started to enforce CLI compliance with the interfaces).

Reply:
> Yes. That does not mean we should make it more broken though, especially for stuff that has very little reason to be broken (of course exposing an IPFS store via the CLI is a lot of low-priority hassle, and those funcs also existed before the refactor that started to enforce CLI compliance with the interfaces).

Agreed, is there a ticket to fix the current broken state or should I make one?

Reply:
Update:
- [x] Removed `ACPModule()`
- Commit: ```PR(ACP): Remove `ACPModule` from `client.Store` ```